jwst/background/background_step.py

```diff
@@ -58,13 +58,38 @@ def process(self, input, bkg_list):
                 result = background_sub.subtract_wfss_bkg(
                                 input_model, bkg_name, wlrange_name)
                 result.meta.cal_step.back_sub = 'COMPLETE'
-
             else:
+                # check if input data is NRS_IFU
+                tolerance = 1.0e-15
+                result = input_model.copy()
+                do_sub = True
+                if input_model.meta.exposure.type in ["NRS_IFU"]:
+                    # check if GWA_XTILT & GWA_YTILT values of source
+                    # background are the same. If not skip step
+                    input_xtilt = input_model.meta.instrument.gwa_xtilt
+                    input_ytilt = input_model.meta.instrument.gwa_ytilt
+                    for bkg_file in bkg_list:
+                        bkg_model = datamodels.ImageModel(bkg_file)
+                        bkg_xtilt = bkg_model.meta.instrument.gwa_xtilt
+                        bkg_ytilt = bkg_model.meta.instrument.gwa_ytilt
+                        xdiff = np.absolute(bkg_xtilt - input_xtilt)
+                        ydiff = np.absolute(bkg_ytilt - input_ytilt)
+                        bkg_model.close()
+                        if xdiff > tolerance or ydiff > tolerance:
+                            do_sub = False
+                            break
                 # Do the background subtraction
-                result = background_sub.background_sub(input_model,
-                                                       bkg_list,
-                                                       self.sigma,
+                if do_sub:
+                    result = background_sub.background_sub(input_model,
+                                                           bkg_list,
+                                                           self.sigma,
                                                        self.maxiters)
-                result.meta.cal_step.back_sub = 'COMPLETE'
+                    result.meta.cal_step.back_sub = 'COMPLETE'
+                else:
+                    print('skip')
+                    result.meta.cal_step.back_sub = 'SKIPPED'
+                    self.log.warning('Skipping background subtraction')
+                    self.log.warning('GWA_XTILT and GWA_YTILT source values '
```

Comment:
Did you intend for these to be the actual keyword names?  The keywords lack the final "T".